build.rs

```diff
@@ -348,13 +348,28 @@ fn pregenerate_asm_main() {
         perlasm(&perlasm_src_dsts, asm_target);
 
         if asm_target.preassemble {
+            // Preassembly is currently only done for Windows targets.
+            assert_eq!(&asm_target.oss, &[WINDOWS]);
+            let os = WINDOWS;
+            let env = "msvc";
+
             if !std::mem::replace(&mut generated_prefix_headers, true) {
                 generate_prefix_symbols_nasm(&pregenerated, &ring_core_prefix()).unwrap();
             }
             let srcs = asm_srcs(perlasm_src_dsts);
+
+            let target = Target {
+                arch: asm_target.arch.to_owned(),
+                os: os.to_owned(),
+                env: env.to_string(),
+                is_git: true,
+                is_debug: false,
```

Comment:
Same question basically 🙂

Reply:
Now `is_git` has been replaced here by `force_warnings_into_errors` which I think is clearly the right thing to do. LMK if you disagree.

Reply:
Regarding `is_debug`, we do want to have the preassembled stuff be a release build as we only really want to support debug building (`NDEBUG` and the like) during development, as `assert()` and similar things can introduce unwanted side channels.